src/hotspot/share/compiler/compileBroker.cpp

```diff
@@ -2224,7 +2223,6 @@ void CompileBroker::invoke_compiler_on_method(CompileTask* task) {
     DTRACE_METHOD_COMPILE_BEGIN_PROBE(method, compiler_name(task_level));
   }
 
-  should_break = directive->BreakAtExecuteOption || task->check_break_at_flags();
```

Comment:
CompileTask::check_break_at_flags is dead code too,  and CIBreakAtOSR, CIBreakAt
they are all defunct right now.

Reply:
I wasn't sure whether or not we wanted to roll the removal of all dead code into this issue or just _break_at_compile. The JBS issue specifically mentions _break_at_compile so that's what I removed here - can expand to do other stuff if you think it's a good idea.

Reply:
I think it's fine to remove all that dead code with this issue

Reply:
Should CIBreakAtOSR and CIBreakAt be removed too, or is it a bug that these flags are ignored?

Reply:
I think the fact that these flags are not working proves that no one is using them and therefore they should be removed.

Reply:
Originally it was used to break into debugger during compilation:
http://hg.openjdk.java.net/jdk8u/jdk8u-dev/hotspot/file/fffb29a53519/src/share/vm/opto/compile.cpp#l2104

In JDK 9 it was replaced with Compiler directive which skips compilation's ciEnv and consult CompilerOracle directly:
https://github.com/openjdk/jdk/blob/master/src/hotspot/share/opto/compile.cpp#L2099

So I think cleaning ciEnv is fine.

I think it is bug that task->check_break_at_flags() is ignored. I used CIBreakAtOSR and CIBreakAt during debugging long ago before directives were introduced. I think it should be fixed.

Reply:
> I think the fact that these flags are not working proves that no one is using them and therefore they should be removed.

I tried to use BreakAtExecute recently, thinking it would generate a break in the generated code, but it didn't work, so I ended up using "break", which if I recall also sets a breakpoint when the method is compiled.  I like having the choice of "break during compile" vs "break in generated code".